kernel/core/extend/pagemem.c

```diff
@@ -59,3 +61,77 @@ void init_pgd()
 	cr0_pg_en(&cr0);
 	set_cr0(cr0);
 }
+
+void init_task_pagemem(tidx task)
+{
+	pde32_t* krn_PGD = nth_pgd_gbl(0);
+	pde32_t* usr_PGD = nth_user_pgds(task);
+
+	switch (task) {
+		case 0:
+			usr_identity_map(&usr_PGD[0], (offset_t)task1, (offset_t)task1 + PTE_OFFSET - 1);   // map task1 code for task1
+			krn_identity_map(&krn_PGD[0], (offset_t)task1, (offset_t)task1 + PTE_OFFSET - 1);   // map task1 code for Kernel
+
+            usr_forced_map(&usr_PGD[0], 0x2000000, SHARED_START, PTE_OFFSET);                   // map shared memory (1 page)
+			break;
+		case 1:
+			usr_identity_map(&usr_PGD[0], (offset_t)task2, (offset_t)task2 + PTE_OFFSET - 1);   // map task2 code for task2
+			krn_identity_map(&krn_PGD[0], (offset_t)task2, (offset_t)task2 + PTE_OFFSET - 1);   // map task2 code for Kernel
+
+            usr_forced_map(&usr_PGD[0], 0x4000000, SHARED_START, PTE_OFFSET);                   // map shared memory (1 page)
+			break;
+		default:
+			break;
+	}
+
+    krn_identity_map(&krn_PGD[0], SHARED_START, SHARED_END + PTE_OFFSET - 1); // map shared usr page for Kernel
+}
+
+void task_forced_map(tidx task, offset_t virtual_address, offset_t physical_address, offset_t size)
+{
+	pde32_t* krn_pgd = nth_pgd_gbl(0);
+	pde32_t* usr_pgd = nth_user_pgds(task);
+
+	// Put in user ptb
+    usr_forced_map(usr_pgd, virtual_address, physical_address, size);
+
+    // Put in kernel ptb
+    krn_identity_map(krn_pgd, physical_address, physical_address + size);
```

Comment:
```suggestion
    krn_identity_map(krn_pgd, physical_address, physical_address + size - 1);
```
end address is included in mapping